samples/nrf9160/asset_tracker/src/main.c

```diff
@@ -882,13 +961,11 @@ void main(void)
 	modem_configure();
 	cloud_connect(NULL);
 
-	if (IS_ENABLED(CONFIG_CLOUD_UA_CONSOLE)) {
-		console_init();
-	}
-
 	while (true) {
 		nrf_cloud_process();
 		input_process();
 		k_sleep(K_MSEC(10));
+		/* Put CPU to idle to save power */
+		k_cpu_idle();
```

Comment:
Still not sure about idling here. I think we'll wake up from idle really fast here, so it's not a big gain in terms of power saving. Could we sleep longer instead or lose this loop altogether and use `poll` somehow to save power? Is that possible when the new MQTT lib is used in `nrf_cloud`, @rlubos ?

Reply:
@jtguggedal : Yes, I think this has to be an update later, when nrf cloud lib is updated to use the upstream MQTT. See PR https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/391

Reply:
First, to clarify - using `poll` is not limited to the upstream implementation. Our local copy is socket-based as well, so we could use `poll` to block the thread rightaway.

The only concern I see for now is the periodic call to the `input_process` function. If we block the thread with `poll`, this function will not be called unless some networking event on a socket occurs, which is not acceptable. We should therefore either create a separate thread for network/input processing or make `input_process` event driven somehow, so instead of calling it periodically we could let the system worqueue process it on some kind of input event. I didn't look in details into the implementation yet, so I'm not sure if it is possible.

Reply:
@rlubos Thanks for the clarification. Then I'd say we should aim to move this to polling. Even though the current implementation of `poll` doesn't actually block, the application will at least be ready for when it eventually does.
I had a quick look, and `input_process` is blocking due to its call to `console_getchar`, so that needs to be replaced by possibly `console_read` and moved to a self-repeating delayed work or, as you say, make it event-driven before we can switch to `poll`. The function should only be called when a device is not paired with an nRF Cloud account and console input of association data is enabled (for devices that don't have buttons and switches), a combination that's probably only used by developers and testers. The repeating work should also disable itself when pairing is complete, so I'd say polling would be acceptable in this case, as it will be enabled for a limited amount of time, and completely disabled when connection to nRF Cloud is established.

Reply:
@jtguggedal Can we do that in separate PR? I consider adding poll to be feature creep to this PR 😄 And it should also be updated in the gateway sample.

Reply:
Sure, it doesn't have to go into this one